stats/stats/src/charts/updater/split.rs

```diff
@@ -1,55 +1,92 @@
-use super::get_min_date_blockscout;
-use crate::{DateValue, UpdateError};
+use super::{get_min_block_blockscout, get_min_date_blockscout, partial::get_last_row};
+use crate::{
+    charts::{find_chart, insert::insert_data_many},
+    Chart, DateValue, UpdateError,
+};
+use async_trait::async_trait;
 use chrono::{Duration, NaiveDate, Utc};
 use sea_orm::{DatabaseConnection, FromQueryResult, Statement, TransactionTrait};
 use std::time::Instant;
 
-pub async fn split_update<F>(
-    blockscout: &DatabaseConnection,
-    last_row: Option<DateValue>,
-    query_maker: F,
-) -> Result<Vec<DateValue>, UpdateError>
-where
-    F: Fn(NaiveDate, NaiveDate) -> Statement,
-{
-    let txn = blockscout
-        .begin()
-        .await
-        .map_err(UpdateError::BlockscoutDB)?;
-    let first_date = match last_row {
-        Some(last_row) => last_row.date,
-        None => get_min_date_blockscout(&txn)
+#[async_trait]
+pub trait ChartSplitUpdater: Chart {
+    fn make_range_query(&self, from_: NaiveDate, to_: NaiveDate) -> Statement;
```

Comment:
Nit: what about `get_query` (to be in line with `get_values` of other updaters) or just `query` (to be more in line with rust conventions)?